packages/nouns-contracts/contracts/governance/NounsDAOInterfaces.sol

```diff
@@ -214,6 +227,176 @@ contract NounsDAOStorageV1 is NounsDAOProxyStorage {
+        uint256 startBlock;
+        /// @notice The block at which voting ends: votes must be cast prior to this block
+        uint256 endBlock;
+        /// @notice Current number of votes in favor of this proposal
+        uint256 forVotes;
+        /// @notice Current number of votes in opposition to this proposal
+        uint256 againstVotes;
+        /// @notice Current number of votes for abstaining for this proposal
+        uint256 abstainVotes;
+        /// @notice Flag marking whether the proposal has been canceled
+        bool canceled;
+        /// @notice Flag marking whether the proposal has been vetoed
+        bool vetoed;
+        /// @notice Flag marking whether the proposal has been executed
+        bool executed;
+        /// @notice Receipts of ballots for the entire set of voters
+        mapping(address => Receipt) receipts;
+        /// @notice The total supply at the time of proposal creation
+        uint256 totalSupply;
+        /// @notice The block at which this proposal was created
+        uint256 creationBlock;
+    }
+
+    /// @notice Ballot receipt record for a voter
+    struct Receipt {
+        /// @notice Whether or not a vote has been cast
+        bool hasVoted;
+        /// @notice Whether or not the voter supports the proposal or abstains
+        uint8 support;
+        /// @notice The number of votes the voter had, which were cast
+        uint96 votes;
+    }
+
+    /// @notice Possible states that a proposal may be in
+    enum ProposalState {
+        Pending,
+        Active,
+        Canceled,
+        Defeated,
+        Succeeded,
+        Queued,
+        Expired,
+        Executed,
+        Vetoed
+    }
+}
+
+/**
+ * @title Storage for Governor Bravo Delegate
+ * @notice For future upgrades, do not change NounsDAOStorageV2. Create a new
+ * contract which implements NounsDAOStorageV2 and following the naming convention
+ * NounsDAOStorageVX.
+ */
+contract NounsDAOStorageV2 is NounsDAOStorageV1Adjusted {
+    DynamicQuorumParamsCheckpoint[] public quorumParamsCheckpoints;
+
+    struct DynamicQuorumParams {
+        /// @notice Minimum quorum votes BPS at the time of proposal creation
+        uint16 minQuorumVotesBPS;
+        /// @notice The maximum basis point number of votes in support of a proposal required in order for a quorum to be reached and for a vote to succeed. *DIFFERS from GovernerBravo
+        uint16 maxQuorumVotesBPS;
+        /// @notice The quorum votes polynom input offset which suppresses polynom contribution until againstVotes.div(totalSupply) reaches this value
+        uint16 quorumVotesBPSOffset;
+        /// @notice Polynomial coefficients for calculating a dynamic quorum based on the amount of against votes
+        /// @dev The coefficients are assumed to be fixed point integer with 6 decimals, i.e 0.2 is represented as 0.2 * 1e6 = 200000. There are 2 coefficients: x^1 and x^2
+        uint32[2] quorumPolynomCoefs;
```

Comment:
We should consider converting the coefficient array to two `uint32`s. Don't see much benefit in the array over two separate variables, and this would allow the struct to pack into one slot, rather than two.

Reply:
Pushed a fix

Reply:
@solimander are you sure it doesn't pack it into one slot?
my understanding is that only dynamic arrays start a new storage slot (from reading [this](https://docs.soliditylang.org/en/v0.8.13/internals/layout_in_storage.html#storage-inplace-encoding))
I was specifically reading this when thinking whether it's ok to keep them as an array in the struct

in any case I don't mind changing to two separate variables, just curious how the storage layout works

Reply:
@davidbrai I read this quote as saying that fixed-size arrays still start a new slot:

> Structs and array data always start a new slot and their items are packed tightly according to these rules.

(dynamic arrays are discussed in a different section just under)

Reply:
I thought the same, but I tested this in remix due to this line:

> Structs and array data always start a new slot and their items are packed tightly according to these rules.

Even with the optimizer on, it takes up two slots. You can see the same if you use `hardhat-storage-layout`.

Reply:
@eladmallel ok I see. I misunderstood it then. thanks!